spec/openapi.yaml

```diff
@@ -53,182 +248,743 @@ paths:
+          description: Updated resource
           content:
             application/json:
               schema:
-                $ref: '#/components/schemas/Error'
+                $ref: '#/components/schemas/Resource'
+              example: |
+                {
+                    'apiVersion': '1.0',
+                    'data': {
+                        'category': 'Tutorial',
+                        'created_at': '2019-07-21 15:41:19',
+                        'downvotes': 0,
+                        'id': 2137,
+                        'languages': [],
+                        'last_updated': '2019-07-22 16:05:53',
+                        'name': 'Some, but not ALL of the knowledge',
+                        'notes': null,
+                        'paid': false,
+                        'times_clicked': 3,
+                        'upvotes': 1,
+                        'url': 'http://www.test.com'
+                    },
+                    'status': 'ok',
+                    'status_code': 200
+                  }
+
+        404:
+          $ref: '#/components/responses/NotFound'
       security:
       - ApiKeyAuth: []
-  /search:
-    get:
-      description: Searches the API for a given keyword
+  /resources/{id}/click:
+    put:
+      summary: Update Resource - Increment Number of Views
+      description:
+        Increments the `times_clicked` property of a resource 
+      parameters:
+      - name: id
+        in: path
+        description: Resource ID
+        required: true
+        style: simple
+        explode: false
+        schema:
+          type: integer
+          format: int64
       responses:
-        200: {}
-  /apikey:
-    post:
-      description: Request API key by providing Operation Code credentials
+        200:
+          description: Updated resource 
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Resource'
+              example: |
+                {
+                  'apiVersion': '1.0',
+                  'data': {
+                      'category': 'Tutorial',
+                      'created_at': '2019-07-21 15:41:19',
+                      'downvotes': 0,
+                      'id': 2137,
+                      'languages': [],
+                      'last_updated': '2019-07-22 16:05:53',
+                      'name': 'Some, but not ALL of the knowledge',
+                      'notes': null,
+                      'paid': false,
+                      'times_clicked': 3,
+                      'upvotes': 1,
+                      'url': 'http://www.test.com'
+                  },
+                  'status': 'ok',
+                  'status_code': 200
+                }
+        404:
+          $ref: '#/components/responses/NotFound'
+
+  /resources:
+    get:
+      summary: List Resources
+      description: Gets list of all available resource entries or filtered entries based on provided criteria.
```

Comment:
So there's a weird quirk that I just remembered about the `List` endpoints vs the `Search` endpoint. The pages are `0` indexed for the search endpoint, but they're `1` indexed from the `List` endpoints. We should call this out in the description section for each of the relevant endpoints so that people writing integrations are aware. Proposed verbiage:
```
Gets list of all available resource entries or filtered entries based on provided criteria. Note: The `page` attribute in the response of the `/search` endpoint begins with `0`, but the `page` attribute in the response of the `List Resources` endpoint begins with `1`.
```

Copy and paste that same thing to all the descriptions for these endpoints, except change the word `Resources` to `Categories` and `Languages` depending on which endpoint you're copying the description into.

Reply:
I left out the part about filtered entries on the 'List Languages' and 'List Categories' since we can't filter at those endpoints.